src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -0,0 +1,280 @@
+
+#nullable enable
+
+using System;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Windows;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Settings;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.ColorSchemes
+{
+    [Export(typeof(IWpfTextViewConnectionListener))]
+    [Export(typeof(ColorSchemeApplier))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Analyzable)]
+    internal sealed partial class ColorSchemeApplier : ForegroundThreadAffinitizedObject, IWpfTextViewConnectionListener, IDisposable
+    {
+        private const string RoslynTextEditorRegistryKey = "Themes\\{de3dbbcd-f642-433c-8353-8f1df4370aba}\\Roslyn Text Editor MEF Items";
+        private const string UseEnhancedColorsSetting = "WindowManagement.Options.UseEnhancedColorsForManagedLanguages";
+        private const string AppliedColorSchemeKeyName = "ColorScheme";
+
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ForegroundColorDefaulter _colorApplier;
+        private readonly ISettingsManager _settingsManager;
+
+        private bool _isDisposed = false;
+        private bool isInitialized = false;
+
+        [ImportingConstructor]
+        [Obsolete]
+        public ColorSchemeApplier(IThreadingContext threadingContext, [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+            : base(threadingContext)
+        {
+            _serviceProvider = serviceProvider;
+            _settingsManager = (ISettingsManager)_serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
+            _colorApplier = new ForegroundColorDefaulter(threadingContext, _serviceProvider);
+        }
+
+        public void Dispose()
+        {
+            // Dispose is invoked when the MEF container is disposed. This will be our
+            // signal that VS is shutting down and we shouldn't try and perform any work.
+            _isDisposed = true;
+        }
+
+        public void SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection<ITextBuffer> subjectBuffers)
+        {
+            AssertIsForeground();
+
+            if (!isInitialized)
+            {
+                isInitialized = true;
+
+                // We need to update the theme whenever the Editor Color Scheme setting changes or the VS Theme changes.
```

Comment:
hmm.. Currently the High Contrast themes are the same for each scheme, which I think is preferable, since we are very limited in our ability to make distinctions using the High Contract system colors. I should make a note though.